### PR TITLE
bf: S3C-3829 Ignore placeholder log entries

### PR DESCRIPTION
**Pull request by naren-rajendran**
Notification populator should not process placeholder log entries created for deletion or cleanup.

### CODE REVIEW

**Review by anurag4DSB — APPROVED** [review]

**Comment by miniscruff** [comment]
Bugfix without a test change?

**Comment by anurag4DSB** [comment]
> Bugfix without a test change?

I thought it will go to integration, right?

**Comment by naren-rajendran** [comment]
> > Bugfix without a test change?
> 
> I thought it will go to integration, right?

Yes.

**Comment by rahulreddy** [comment]
> > > Bugfix without a test change?
> > 
> > 
> > I thought it will go to integration, right?
> 
> Yes.

There should be a test here for this (unit/functional/both), we can't wait for a Integration test to give us feedback if this is fixed or not. I understand having a test setup here is probably hard but we should not use that as an excuse to defer testing in Integration.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Comment by naren-rajendran** [comment]
> > > > Bugfix without a test change?
> > > 
> > > 
> > > I thought it will go to integration, right?
> > 
> > 
> > Yes.
> 
> There should be a test here for this (unit/functional/both), we can't wait for a Integration test to give us feedback if this is fixed or not. I understand having a test setup here is probably hard but we should not use that as an excuse to defer testing in Integration.

I refactored backbeat to handle this in util function and added a test to it.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Review by jonathan-gramain — APPROVED** [review]

File: extensions/notification/utils/config.js [hunk 1]
```diff
@@ -69,6 +69,14 @@ function filterConfigsByEvent(bnConfigs, event) {
  */
 function validateEntry(bnConfig, entry) {
     const { bucket, eventType } = entry;
+    /**
+     * if the event type is undefined, it is most likely an entry that is a
+     * placeholder for deletion or cleanup, these entries should be ignored and
+     * not be processed.
```
Comment (miniscruff) [inline extensions/notification/utils/config.js]: * You say "is undefined" but use a falsey check below. Should this be `eventType === undefined`?
* "most likely" is not a very concrete description, it sounds like there is a chance it's something else. What that else is I have no idea.

Comment (naren-rajendran) [reply extensions/notification/utils/config.js]: corrected.

**Comment by naren-rajendran** [comment]
@bert-e reset

**Review by miniscruff — APPROVED** [review]

**Review by rahulreddy — APPROVED** [review]

**Pull request merged**